usaspending_api/common/serializers.py

```diff
@@ -7,83 +7,127 @@ class LimitableSerializer(serializers.ModelSerializer):
-        except AttributeError:
-            # We don't have any nested serializers
-            pass
 
-        req = self.context.get('req')
-        request = self.context.get('request')
-
-        if request:
-            params = get_params_from_req_or_request(request=request, req=req)
-
-            exclude_fields = params.get('exclude')
-            include_fields = params.get('fields')
-            current_viewset = self.context.get('view')
-
-            if params.get('verbose', False):
-                # We have a request for verbose, so we return here so that we
-                # return all fields
-                return
-
-            # We must exclude before include to avoid conflicts from user error
-            if exclude_fields is not None and not self.context.get("child", False):  # the child check should be removed when child field limiting is implemented
-                for field_name in exclude_fields:
-                    try:
-                        self.fields.pop(field_name)
-                    except KeyError:
-                        # Because we're not currently handling nested serializer field
-                        # limiting pass-down, this can happen due to the context pass down
-                        pass
-
-            if include_fields is not None and not self.context.get("child", False):  # the child check should be removed when child field limiting is implemented
-                allowed = set(include_fields)
-                existing = set(self.fields.keys())
-                for field_name in existing - allowed:
-                    try:
-                        # If we have a coded field, always include its description
-                        if field_name.split("_")[-1] == "description" and "_".join(field_name.split("_")[:-1]) not in allowed:
-                            continue
-                        self.fields.pop(field_name)
-                    except KeyError:
-                        # Because we're not currently handling nested serializer field
-                        # limiting pass-down, this can happen due to the context pass down
-                        pass
-
-            elif current_viewset and current_viewset.action == 'retrieve':
-                # The view has specifically asked that all fields should
-                # be returned (for example, when the request url follows
-                # the item/{pk} pattern)
-                return
-
-            else:
-                try:
-                    include_fields = self.Meta.model.get_default_fields(path=request._request.path_info)
-                    allowed = set(include_fields)
-                    existing = set(self.fields.keys())
-                    for field_name in existing - allowed:
-                        self.fields.pop(field_name)
-                except AttributeError:
-                    # We don't have get default fields available
-                    pass
+        # Now we alter our own field sets
```

Comment:
It's great to see the default field logic de-coupled from the models. Since the trade-off for doing so (and a worthwhile one, imo) is to add some complexity to the serializer, does it make sense to encapsulate the logic above hat ultimately produces the final `include_fields` and `exclude_fields` list into a function that can be unit-tested?

Where I'm going with this thought is giving us a low barrier to entry for testing various permutations of the `**kwargs` and `default_fields` options to ensure that the resulting include/exclude field lists are what we expect.